src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs

```diff
@@ -0,0 +1,230 @@
+
+                internal override int? Glyph => (int)CodeAnalysis.Glyph.Reference;
+
+                private readonly Lazy<string> _lazyResolvedPath;
+
+                public AssemblyReferenceCodeAction(
+                    AssemblyReference reference,
+                    Document document,
+                    SyntaxNode node,
+                    bool placeSystemNamespaceFirst)
+                {
+                    _reference = reference;
+                    _document = document;
+                    _node = node;
+                    _placeSystemNamespaceFirst = placeSystemNamespaceFirst;
+
+                    _title = $"{reference.provider.GetDescription(reference.SearchResult.NameParts)} ({string.Format(FeaturesResources.from_0, reference._assemblyName)})";
+                    _lazyResolvedPath = new Lazy<string>(ResolvePath);
+                }
+
+                private string ResolvePath()
+                {
+                    var metadataService = _document.Project.Solution.Workspace.Services.GetService<IAssemblyPathResolver>();
+                    var assemblyPath = metadataService?.ResolveAssemblyPath(_document.Project.Id, _reference._assemblyName);
+                    return assemblyPath;
+                }
+
+                internal override bool PerformFinalApplicabilityCheck => true;
+
+                internal override bool IsApplicable(Workspace workspace)
+                {
+                    return !string.IsNullOrWhiteSpace(_lazyResolvedPath.Value);
+                }
+
+                protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
+                {
+                    var service = _document.Project.Solution.Workspace.Services.GetService<IMetadataService>();
+                    var resolvedPath = _lazyResolvedPath.Value;
+                    var reference = service.GetReference(resolvedPath, MetadataReferenceProperties.Assembly);
+
+                    // First add the "using/import" directive in the code.
+                    var node = _node;
+                    var document = _document;
+                    _reference.ReplaceNameNode(ref node, ref document, cancellationToken);
+
+                    var newDocument = await _reference.provider.AddImportAsync(
+                        node, _reference.SearchResult.NameParts, document, _placeSystemNamespaceFirst, cancellationToken).ConfigureAwait(false);
+
+                    // Now add the actual assembly reference.
+                    var newProject = newDocument.Project;
+                    newProject = newProject.WithMetadataReferences(
+                        newProject.MetadataReferences.Concat(reference));
+
+                    var operation = new ApplyChangesOperation(newProject.Solution);
+                    return SpecializedCollections.SingletonEnumerable<CodeActionOperation>(operation);
+                }
+
+#if false
```

Comment:
Yeah.  I cloned it from the NugetReference type.  I then iifdefed out all the code i didn't need.  Will remove.